x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRestoreSourceService.java

```diff
@@ -103,7 +104,9 @@ public synchronized Store.MetadataSnapshot openSession(String sessionUUID, Index
                 if (indexShard.state() == IndexShardState.CLOSED) {
                     throw new IndexShardClosedException(indexShard.shardId(), "cannot open ccr restore session if shard closed");
                 }
-                restore = new RestoreSession(sessionUUID, indexShard, indexShard.acquireSafeIndexCommit(), scheduleTimeout(sessionUUID));
+                final Engine.IndexCommitRef commitRef = indexShard.acquireSafeIndexCommit();
+                final Set<String> fileNames = Set.copyOf(commitRef.getIndexCommit().getFileNames());
+                restore = new RestoreSession(sessionUUID, indexShard, commitRef, fileNames, scheduleTimeout(sessionUUID));
```

Comment:
Storing the known list of file names on `RestoreSession` creation time so that later checks for known file name is lightweight. 
Thanks to @Tim-Brooks for the suggestion!